### ISSUE TITLE
bug on Admin webtools panel

### ISSUE THREAD

**Issue by alnux** [phpMv/ubiquity#202]
hi i just install ubiquity 2.5.0 and after using "Ubiquity serve" command line i go to the browser when i push webtools link (http://localhost/Admin) it show me a error with foreach() argument. [here a screenshot ](https://prnt.sc/Kny1Iq4wtJg5) and [here a screenshot](https://prnt.sc/KbAnjEqxdPN-) of the error line im using php 8.1.12.

regards

**Comment by jcheron** [comment]
Hi @alnux 
Can you add to your original message the result of 
`Ubiquity version`
executed from the root of your project:
>    · PHP 8.1.1
>    · Ubiquity devtools (1.4.0)
>    · Ubiquity webtools (2.5.0)
>    · Ubiquity 2.5.0

Is this a brand new project, or a migration on an existing one?

**Comment by alnux** [comment]
· The project folder is /Volumes/ALLEN-FILES/UBIQUITY/PROYECTS/firstProject

    · PHP 8.1.12
    · Ubiquity devtools (1.4.0)
    · Ubiquity webtools (2.5.0)
    · Ubiquity 2.5.0
    · OS: Darwin imac-de-stream 19.6.0 Darwin Kernel Version 19.6.0: Tue Nov 10 00:10:30 PST 2020; root:xnu-6153.141.10~1/RELEASE_X86_64 x86_64

**Comment by alnux** [comment]
by the way im trying to use with apache with vhost instead Ubiquity serve but there is a problem with .htaccess wich cause next error
`[Wed Dec 21 23:01:13.804061 2022] [core:error] [pid 3001] [client 127.0.0.1:50310] AH00124: Request exceeded the limit of 10 internal redirects due to probable configuration error. Use 'LimitInternalRecursion' to increase the limit if necessary. Use 'LogLevel debug' to get a backtrace., referer: http://proyecto.alnux/`

my generated htaccess is
///////////////////////////////////////////////////////////// i dont know why github put like this
AddDefaultCharset UTF-8
<IfModule mod_rewrite.c>
	RewriteEngine On
	RewriteBase /firstProject/public/
	
	RewriteCond %{REQUEST_FILENAME} !-f  
	RewriteCond %{HTTP_ACCEPT} !(.*images.*)
	RewriteRule ^(.*)$ index.php?c=$1 [L,QSA]
</IfModule>
////////////////////////////////////////////////////////////////////////////////

but if i comment all htaccess directives, the apache error log show me the next error

`[Wed Dec 21 23:18:51.363955 2022] [php:warn] [pid 3843] [client 127.0.0.1:50545] PHP Warning:  include(/Volumes/ALLEN-FILES/UBIQUITY/PROYECTS/firstProject/public/../app/cache/config/config.cache.php): Failed to open stream: No such file or directory in /Volumes/ALLEN-FILES/UBIQUITY/PROYECTS/firstProject/public/index.php on line 4`

`[Wed Dec 21 23:18:51.364151 2022] [php:warn] [pid 3843] [client 127.0.0.1:50545] PHP Warning:  include(): Failed opening '/Volumes/ALLEN-FILES/UBIQUITY/PROYECTS/firstProject/public/../app/cache/config/config.cache.php' for inclusion (include_path='.:/Applications/XAMPP/xamppfiles/lib/php') in /Volumes/ALLEN-FILES/UBIQUITY/PROYECTS/firstProject/public/index.php on line 4`

`[Wed Dec 21 23:18:51.367380 2022] [php:error] [pid 3843] [client 127.0.0.1:50545] PHP Fatal error:  Uncaught TypeError: Ubiquity\\cache\\CacheManager::startProd(): Argument #1 ($config) must be of type array, bool given, called in /Volumes/ALLEN-FILES/UBIQUITY/PROYECTS/firstProject/app/config/services.php on line 4 and defined in /Volumes/ALLEN-FILES/UBIQUITY/PROYECTS/firstProject/vendor/phpmv/ubiquity/src/Ubiquity/cache/CacheManager.php:44\nStack trace:\n#0 /Volumes/ALLEN-FILES/UBIQUITY/PROYECTS/firstProject/app/config/services.php(4): Ubiquity\\cache\\CacheManager::startProd(false)\n#1 /Volumes/ALLEN-FILES/UBIQUITY/PROYECTS/firstProject/public/index.php(6): require('/Volumes/ALLEN-...')\n#2 {main}\n  thrown in /Volumes/ALLEN-FILES/UBIQUITY/PROYECTS/firstProject/vendor/phpmv/ubiquity/src/Ubiquity/cache/CacheManager.php on line 44`

on the last report it talk about an that has to be array but bool is given.
on the first two errors the cache directory and all subdirectories has the write permisions to auto create the files that needs, but them not autocreate, the /app/cache/config/ is still empty

**Comment by alnux** [comment]
hello I found the error,maybe you forgot to correct the path after doing your tests, but in the index.php file I corrected the line 4.
from
`$config = include ROOT . 'cache/config/config.cache.php';`
to
`$config = include ROOT . 'config/config.php';`

I don't know if it is the new configuration or not, but it doesn't work with the cache directory, as I told you there is nothing inside the indicated folder in new project.

now works with apache [screenshot](https://prnt.sc/mqQKZjrptiMw)

is a great project

**Comment by jcheron** [comment]
With `Ubiquity 2.5.0`
```php
ROOT . 'cache/config/config.cache.php'
```
is the new config file location, generated with `Ubiquity init-cache`

In your case, the configuration cache was not generated automatically (as it should have been when the project was created).
This bug may be due to the fact that your globally installed devtools are not up to date.

Can you try to update the devtools:
```bash
composer global update
```
and then create a new project to see if it's better?

**Comment by alnux** [comment]
hi there again,
well i made a 
`composer global update`

create new project with the command that is on guide page
`Ubiquity new firstProject -a`

and i had the same error
`../app/cache/config/config.cache.php)`

but using `Ubiquity init-cache` all was ok with `Ubiquity serve` but with apache there was an error 500 this was becouse on .htaccess is a rewriteBase rule `RewriteBase /firstProject/public/` wich i comment and the first page appears, after i push webtools button and it goes to `http://127.0.0.1/firstProject/public/` wich i modificated too and after that all was ok.

the thing is that the `Ubiquity init-cache` does not autorun after installation, at least in macos composer.

**Comment by jcheron** [comment]
Fixed in 2.5.1
